modules/platforms/cpp/core-test/src/cache_query_schema_test.cpp

```diff
@@ -0,0 +1,311 @@
+
+        Sql("DROP TABLE " + TableName(pred()));
+    }
+
+    /** Node started during the test. */
+    Ignite grid;
+};
+
+BOOST_FIXTURE_TEST_SUITE(CacheQuerySchemaTestSuite, CacheQuerySchemaTestSuiteFixture)
+
+bool TruePred()
+{
+    return true;
+}
+
+BOOST_AUTO_TEST_CASE(TestBasicOpsExplicitPublicSchema)
+{
+    ExecuteStatementsAndVerify(TruePred);
+}
+
+bool FalsePred()
+{
+    return false;
+}
+
+BOOST_AUTO_TEST_CASE(TestBasicOpsImplicitPublicSchema)
+{
+    ExecuteStatementsAndVerify(FalsePred);
+}
+
+struct MixedPred
+{
+    int i;
+
+    MixedPred() : i(0)
+    {
+        // No-op.
+    }
+
+    bool operator()()
+    {
+        return (++i & 1) == 0;
+    }
+};
+
+BOOST_AUTO_TEST_CASE(TestBasicOpsMixedPublicSchema)
+{
+    MixedPred pred;
+
+    ExecuteStatementsAndVerify(pred);
+}
+
+BOOST_AUTO_TEST_CASE(TestCreateDropNonExistingSchema)
+{
+    BOOST_CHECK_THROW(
+        Sql("CREATE TABLE UNKNOWN_SCHEMA." + TABLE_NAME + "(id INT PRIMARY KEY, val INT)"),
+        IgniteError
+    );
+
+    BOOST_CHECK_THROW(
+        Sql("DROP TABLE UNKNOWN_SCHEMA." + TABLE_NAME),
+        IgniteError
+    );
+}
+
+BOOST_AUTO_TEST_CASE(TestBasicOpsDiffSchemas)
+{
+    Sql("CREATE TABLE " + SCHEMA_NAME_1 + '.' + TABLE_NAME + " (s1_key INT PRIMARY KEY, s1_val INT)");
+    Sql("CREATE TABLE " + SCHEMA_NAME_2 + '.' + TABLE_NAME + " (s2_key INT PRIMARY KEY, s2_val INT)");
+    Sql("CREATE TABLE " + Q_SCHEMA_NAME_3 + '.' + TABLE_NAME + " (s3_key INT PRIMARY KEY, s3_val INT)");
+    Sql("CREATE TABLE " + SCHEMA_NAME_4 + '.' + TABLE_NAME + " (s4_key INT PRIMARY KEY, s4_val INT)");
+
+    Sql("INSERT INTO " + SCHEMA_NAME_1 + '.' + TABLE_NAME + " (s1_key, s1_val) VALUES (1, 2)");
+    Sql("INSERT INTO " + SCHEMA_NAME_2 + '.' + TABLE_NAME + " (s2_key, s2_val) VALUES (1, 2)");
+    Sql("INSERT INTO " + Q_SCHEMA_NAME_3 + '.' + TABLE_NAME + " (s3_key, s3_val) VALUES (1, 2)");
+    Sql("INSERT INTO " + SCHEMA_NAME_4 + '.' + TABLE_NAME + " (s4_key, s4_val) VALUES (1, 2)");
+
+    Sql("UPDATE " + SCHEMA_NAME_1 + '.' + TABLE_NAME + " SET s1_val = 5");
+    Sql("UPDATE " + SCHEMA_NAME_2 + '.' + TABLE_NAME + " SET s2_val = 5");
+    Sql("UPDATE " + Q_SCHEMA_NAME_3 + '.' + TABLE_NAME + " SET s3_val = 5");
+    Sql("UPDATE " + SCHEMA_NAME_4 + '.' + TABLE_NAME + " SET s4_val = 5");
+
+    Sql("DELETE FROM " + SCHEMA_NAME_1 + '.' + TABLE_NAME);
+    Sql("DELETE FROM " + SCHEMA_NAME_2 + '.' + TABLE_NAME);
+    Sql("DELETE FROM " + Q_SCHEMA_NAME_3 + '.' + TABLE_NAME);
+    Sql("DELETE FROM " + SCHEMA_NAME_4 + '.' + TABLE_NAME);
+
+    Sql("CREATE INDEX t1_idx_1 ON " + SCHEMA_NAME_1 + '.' + TABLE_NAME + "(s1_val)");
+    Sql("CREATE INDEX t1_idx_1 ON " + SCHEMA_NAME_2 + '.' + TABLE_NAME + "(s2_val)");
+    Sql("CREATE INDEX t1_idx_1 ON " + Q_SCHEMA_NAME_3 + '.' + TABLE_NAME + "(s3_val)");
+    Sql("CREATE INDEX t1_idx_1 ON " + SCHEMA_NAME_4 + '.' + TABLE_NAME + "(s4_val)");
+
+    Sql("SELECT * FROM " + SCHEMA_NAME_1 + '.' + TABLE_NAME);
```

Comment:
Well, again, I've just took Java tests and implemented them in C++. I've thought that this is going to be enough since they were checked and accepted.